pkg/gadgets/top/ebpf/tracer/tracer.go

```diff
@@ -325,7 +322,15 @@ func (t *Tracer) nextStats() ([]*types.Stats, error) {
 			CumulativeRunCount: cumulativeRunCount,
 			MapMemory:          totalMapMemory,
 			MapCount:           uint32(len(mapIDs)),
-		})
+		}
+
+		if t.enricher != nil {
+			// We are only interested in node here, so giving 0 as MountNsID should be
+			// fine.
+			t.enricher.Enrich(&stat.CommonData, 0)
```

Comment:
What about adding a check in the Enricher to avoid iterating unnecessary through all the containers in `LookupContainerByMntns`?

```golang
func (cc *ContainerCollection) EnrichByMntNs(event *eventtypes.CommonData, mountnsid uint64) {
	event.Node = cc.nodeName

	if mountnsid == 0 {
		log.Debug("mount namespace is 0, enriching event with node only")
		return
	}

	container := cc.LookupContainerByMntns(mountnsid)
	[...]
}

```